test: revive and harden E2E test suite for simulator

yilmazbahadir opened the pull request:
### Summary
This PR revives the stale E2E test suite and hardens it to run reliably with the Lattice simulator.

### Key Changes
- **Test Setup**: Refactor tests to use `beforeAll` for consistent client pairing
- **CI Compatibility**: Skip SafeCard-dependent tests in CI where physical cards aren't available
- **Signing & API**: Support raw transaction inputs and improve reliability
- **Bug Fixes**: 
  - Normalize legacy KV transactions
  - Fix SafeCard non-exportable test hash
  - Improve contract test stability

### Testing
All E2E tests now pass with the simulator.
```
CI=1 DEBUG_SIGNING=1 baseUrl=http://127.0.0.1:3000 DEVICE_ID=SD0001 PASSWORD=12345678 PAIRING_SECRET=12345678 ENC_PW=12345678 APP_NAME=lattice-manager pnpm vitest run --silent --reporter=basic
```
<img width="953" height="442" alt="image" src="https://github.com/user-attachments/assets/513a66d0-9780-4717-b85a-a53d2cf66ba0" />

yilmazbahadir committed:
feat: improve eip-712 and generic signing reliability

- Fix EIP712 integer type handling with proper bit-width calculation
- Enhance BigInt precision preservation in ethereum.ts
- Improve parseGenericSigningResponse with digest support and better offset tracking
- Add computeMessageHash helper for robust hash computation
- Fix test descriptions and error message expectations
- Better two's complement handling for signed integers

These changes improve the reliability of EIP712 message signing
and generic signing operations, particularly for large integer values.

yilmazbahadir committed:
test: fix bitcoin test setup to use beforeAll

- Change test('pair') to beforeAll() for proper test setup
- Ensures client is initialized before running Bitcoin tests
- Prevents test isolation issues with client setup

yilmazbahadir committed:
test: fix test setup and add CI skip conditions

yilmazbahadir committed:
fix: normalize legacy kv transactions

Coerce hex nonce values and default empty data so legacy transaction payloads continue to parse after the viem schema tightenings.

Also bounds the kv cleanup helper in the e2e test to avoid reissuing the same getKvRecords request indefinitely.

yilmazbahadir committed:
fix: safe card non-exportable test hash

yilmazbahadir committed:
feat(signing): support raw tx inputs and harden api e2e

- Allow sign to accept pre-serialized hex/buffer payloads while preserving EIP-7702 encoding detection and decoder enrichment
- Auto-detect EIP-712 message payloads in signMessage so callers only supply the typed data structure
- Stabilise API e2e tests by mocking decoder calls, suppressing noisy logs, reusing the paired client, and exercising address-tag flows against the simulator after a connectivity probe

yilmazbahadir committed:
test: improve contract test reliability and forge integration

- Add forge CLI availability check to conditionally skip tests
- Replace static import with dynamic artifact loading from forge output
- Improve error handling and logging for missing forge dependencies
- Make contract tests more robust in CI environments

yilmazbahadir committed:
test: add xpub test and improve wallet-jobs test setup

- Add xpub test for Bitcoin extended public key derivation
- Refactor test setup to use beforeAll for consistent pairing
- Skip SafeCard-dependent tests in CI environment
- Remove redundant pairing checks from wallet-jobs tests

yilmazbahadir committed:
fix: lint

yilmazbahadir committed:
refactor: improve type safety in signing module

Add explicit type annotations for encodingType variable to include EIP7702
encoding types and add type assertion for basePayload to improve type safety
and IDE support.

netbonus commented on src/__test__/e2e/contracts.test.ts:
in a follow-up PR we can migrate these tests to the viem signer validation system and remove forge entirely (or we might already cover these test cases with other viem signer comparisons and we can remove now).

yilmazbahadir replied:
captured an issue: https://github.com/GridPlus/gridplus-sdk/issues/623

netbonus commented on src/__test__/e2e/general.test.ts:
```diff
@@ -40,11 +40,15 @@ const id = getDeviceId();
 
-  it('Should test SDK dehydration/rehydration', async () => {
+  it('Should test SDK dehydration/rehydration', async (ctx) => {
+    if (process.env.CI === '1') {
```
why skip in CI?

yilmazbahadir replied:
it works in the CI too, re-enabled 👍

netbonus commented on src/__test__/e2e/non-exportable.test.ts:
```diff
@@ -171,7 +176,32 @@ describe('Non-Exportable Seed', () => {
 });
 
-function validateSig(resp: any, hash: Buffer) {
+function toBuffer(data: string | Buffer | Uint8Array): Buffer {
```
these utils seem like they shouldn't be in a test file?

yilmazbahadir replied:
good call, I've extracted to `gridplus-sdk/src/__test__/utils/helpers.ts`

yilmazbahadir committed:
test: enable sdk dehydration/rehydration test in ci

Remove CI skip condition to allow the test to run in continuous integration
environments.

yilmazbahadir committed:
refactor(test): extract signature validation helpers to shared module

Move signature validation utilities from non-exportable.test.ts to helpers.ts
to improve code reusability and reduce duplication.

Changes:
- Extract toBuffer, toUint8Array, ensureHash32, and validateSig to helpers.ts
- Enhance validateSig to properly handle compressed/uncompressed public keys
- Remove duplicate helper functions from test file
- Update imports to use shared utilities
